### PR TITLE
Adding test scripts (YAMLs) for Laundry Dryer Mode

### PR DESCRIPTION
**Pull request by prakashece**
The test scripts are based on LaundryDryerControls.adoc. Please find the link below to access it.

https://github.com/CHIP-Specifications/chip-test-plans/pull/3686

### CODE REVIEW

**Comment by woody-apple** [comment]
@prakashece Please check restyle, it's complaining about the format of the YAML document.

**Comment by prakashece** [comment]
> @prakashece Please check restyle, it's complaining about the format of the YAML document.

@woody-apple Fixed the restyle issue.

**Review by manjunath-grl — CHANGES_REQUESTED** [review]
@prakashece The script should begin with the inclusion of CHIP copyrights.
https://github.com/project-chip/connectedhomeip/blob/fb47c905f96e348976612105faa5ef04654e8888/src/app/tests/suites/certification/Test_TC_MOD_3_2.yaml#L1

**Comment by manjunath-grl** [comment]
Test case IDs should adhere to the format, for example: Test_TC_<TestcaseName>_ID.yaml. Additionally, automated test cases must be included in [ciTests.json](https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites/ciTests.json)

**Comment by prakashece** [comment]
> Test case IDs should adhere to the format, for example: `Test_TC_<TestName>_ID.yaml`. Additionally, automated test cases must be included in [ciTests.json](https://github.com/project-chip/connectedhomeip/blob/master/src/app/tests/suites/ciTests.json)

@manjunath-grl addressed your comments. ciTests.json will be pushed in another Pull request

**Review by raju-apple — CHANGES_REQUESTED** [review]
1. @prakashece Why do we have 2 copies of the YAMLs , can we delete the other one not adhering to the format ? 
2. Please add the new tests also in the ciTests.json 
Here's a sample PR for reference https://github.com/project-chip/connectedhomeip/pull/28985

**Comment by prakashece** [comment]
> 1. @prakashece Why do we have 2 copies of the YAMLs , can we delete the other one not adhering to the format ?
> 2. Please add the new tests also in the ciTests.json
>    Here's a sample PR for reference [Modified tests Aug 31 #28985](https://github.com/project-chip/connectedhomeip/pull/28985)

@raju-apple Deleted the other 2 files and added the tests in ciTests.json. Please check and let me know if you have any comments

**Review by raju-apple — APPROVED** [review]

File: src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml [hunk 1]
```diff
@@ -0,0 +1,97 @@
...
+
+PICS:
+    - DRYERCTRL.S
+
+config:
+    nodeId: 0x12344321
+    cluster: "Laundry Dryer Controls"
+    endpoint: 1
+
+unsupportedSelectedDrynessLevel:
```
Comment (manjunath-grl) [inline src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml:25]: Here Indentation is wrong.

**Review by manjunath-grl — APPROVED** [review]

**Review by manjunath-grl — APPROVED** [review]

File: src/app/tests/suites/certification/TC-DRYERCTRL-1.1.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+
+    - label: "Step2: Read the global attribute: ClusterRevision"
+      command: "readAttribute"
+      attribute: "ClusterRevision"
+      response:
+          value: 1
+          constraints:
+              type: int16u
+
+    - label: "Step 3: Read the global attribute: FeatureMap"
```
[the remark this answers is not in the document]
Comment (prakashece) [reply src/app/tests/suites/certification/TC-DRYERCTRL-1.1.yml]: @manjunath-grl updated the test scripts

**Review by manjunath-grl — APPROVED** [review]

**Review by manjunath-grl — APPROVED** [review]

File: src/app/tests/suites/certification/PICS.yaml [hunk 3]
```diff
@@ -8877,6 +8877,53 @@ PICS:
...
 
+    #  Laundry DRYER Mode Cluster
+
+    - label: "Does the device implement the Dryer cluster as a server?"
+      id: DRYERCTRL.S
+
+    - label: "Does the DUT support testing the failed ChangeToMode command?"
+      id: DRYERCTRL.S.M.CAN_TEST_MODE_FAILURE
+
+    - label:
```
Comment (cecille) [inline src/app/tests/suites/certification/PICS.yaml:8888]: I don't think this file is meant to have pixits in it.

File: src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml [hunk 4]
```diff
@@ -0,0 +1,102 @@
+# Copyright (c) 2021 Project CHIP Authors
```
Comment (cecille) [inline src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml:1]:
```suggestion
# Copyright (c) 2023 Project CHIP Authors
```

File: src/app/tests/suites/certification/Test_TC_DRYERCTRL_1_1.yaml [hunk 5]
```diff
@@ -0,0 +1,91 @@
+# Copyright (c) 2021 Project CHIP Authors
```
Comment (cecille) [inline src/app/tests/suites/certification/Test_TC_DRYERCTRL_1_1.yaml:1]:
```suggestion
# Copyright (c) 2023 Project CHIP Authors
```

File: src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml [hunk 6]
```diff
@@ -0,0 +1,102 @@
...
+
+PICS:
+    - DRYERCTRL.S
+
+config:
+    nodeId: 0x12344321
+    cluster: "Laundry Dryer Controls"
+    endpoint: 1
+
+    unsupportedSelectedDrynessLevel:
```
Comment (cecille) [inline src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml:25]: I know this is more of a test plan thing, but is the device guaranteed to HAVE an unsupported selected dryness level?

File: src/app/tests/suites/certification/ci-pics-values [hunk 7]
```diff
@@ -750,6 +750,25 @@ PIXIT.LWM.MODE_CHANGE_OK=1
...
+DRYERCTRL.S.A0001=1
+
+#Feature
+DRYERCTRL.S.F00=1
+
+#Command
+DRYERCTRL.S.C00.Rsp=1
+DRYERCTRL.S.C01.Tx=1
+
+#PIXIT
```
Comment (cecille) [inline src/app/tests/suites/certification/ci-pics-values:766]: These aren't required here.

**Review by cecille — APPROVED** [review]

File: src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml (hunk 6 above)
Comment (prakashece) [reply src/app/tests/suites/certification/Test_TC_DRYERCTRL_2_1.yaml:25]: It doesn't have an unsupported dryness level it is just for the test case

Just to check if it doesn't accept an unsupported dryness level

File: src/app/tests/suites/certification/ci-pics-values (hunk 7 above)
Comment (prakashece) [reply src/app/tests/suites/certification/ci-pics-values:762]: updated

File: src/app/tests/suites/certification/PICS.yaml (hunk 3 above)
Comment (prakashece) [reply src/app/tests/suites/certification/PICS.yaml]: Removed the Pixits

**Review by jmartinez-silabs — APPROVED** [review]

**Comment by prakashece** [comment]
Created a new one https://github.com/project-chip/connectedhomeip/pull/31250

**Pull request closed without merging**